numba/targets/arraymath.py

```diff
@@ -765,6 +765,109 @@ def median_impl(arry):
 
     return median_impl
 
+@register_jitable
+def _collect_percentiles(a, q, skip_nan=False):
+
+    if np.any(np.isnan(q)) or np.any(q < 0) or np.any(q > 100):
+        raise ValueError('Percentiles must be in the range [0,100]')
+
+    temp_arry = a.flatten()  # use as temp workspace; may mutate
+    nan_mask = np.isnan(temp_arry)
+
+    # check elements of supplied array, a
+    if skip_nan:
+        # if we're told to skip NaNs, but there are no data points left, exit
+        temp_arry = temp_arry[~nan_mask]
+        if len(temp_arry) == 0:
+            return np.full(len(q), np.nan)
+    else:
+        # if we're told to not skip NaNs, but we find one, exit
+        if np.any(nan_mask):
+            return np.full(len(q), np.nan)
+
+    n = len(temp_arry)
+    if n == 1:
+        # single element array - return value same for all q
+        val = temp_arry[0]
+        if np.isfinite(val):
+            fill_value = val
+        else:
+            fill_value = np.nan
+        out = np.full(len(q), fill_value, dtype=np.float64)
+    else:
+        out = np.empty(len(q), dtype=np.float64)
+
+        for i in range(len(q)):
+            percentile = q[i]
+            if percentile == 100:
+                val = np.max(temp_arry)  # bypass pivoting
+            elif percentile == 0:
+                val = np.min(temp_arry)  # bypass pivoting
+            else:
+                rank = 1 + (n - 1) * np.true_divide(percentile, 100.0)  # linear interp between closest ranks
+                f = math.floor(rank)
```

Comment:
Out of interest, why `math.floor` and not `np.floor`?

Reply:
I noticed that math.floor is used elsewhere in arraymath.py - I thought I would follow suit, but happy to change.